### PR TITLE
Remove a bunch of jsc files that should have been removed with 58

### PR DESCRIPTION
**Pull request by acoates-ms**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and Microsoft/react-native :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into Microsoft/react-native :thumbsup:
- [ ] I am making a fix / change for the macOS implementation of react-native
- [x] I am making a change required for Microsoft usage of react-native

#### Description of changes

Remove a bunch of files that were removed in RN 0.58

#### Focus areas to test

Get things building

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/Microsoft/react-native/pull/44)

### CODE REVIEW

File: ReactCommon/jsi/sources.inc [hunk 1]
```diff
@@ -14,22 +14,10 @@ SOURCES_V8_FILES = \
   ..\V8Runtime_droid.cpp \
   ..\V8Runtime_shared.cpp \
 
-SOURCES_JSC_FILES = \
-  ..\JSCRuntime.cpp \
```
Comment (rohitjain85) [inline ReactCommon/jsi/sources.inc]: We have kept this because we wanted to keep the ability to switch b/w JS Engines.

Comment (acoates-ms) [reply ReactCommon/jsi/sources.inc]: The jsc files were deleted from react native as part of RN 58.  It seems unlikely that we would readd the jsc impl over the V8 or other Js engines.

I am cleaning this up to unblock devmain which hit a break caused by the deletion of some of the jsc files.

Comment (rohitjain85) [reply ReactCommon/jsi/sources.inc]: I am fine with cleaning jsc executor flow but JSCRuntime is present in latest RN and we should be able to conditionally fallback to it.

---
In reply to: [273331168](https://github.com/Microsoft/react-native/pull/44#discussion_r273331168) [](ancestors = 273331168)

Comment (rohitjain85) [reply ReactCommon/jsi/sources.inc]: basically keep things as-is in ReactCommon\jsi

---
In reply to: [273334702](https://github.com/Microsoft/react-native/pull/44#discussion_r273334702) [](ancestors = 273334702,273331168)

Comment (rohitjain85) [reply ReactCommon/jsi/sources.inc]: I am fine with cleaning jsc executor flow but JSCRuntime is present in latest RN and we should be able to conditionally fallback to it.
basically keep things as-is in ReactCommon\jsi

**Comment by rohitjain85** [comment]
We need to verify that V8Executor flow is intact as that is what we use in devmain.
It can be verified by ensuring RN Tester is running well after commenting out getJavaScriptExecutorFactory in RNTesterApplication.java (commenting the method will ensure that we are not going to use JSI-V8Runtime flow)

**Comment by rohitjain85** [comment]
>   return folly::make_unique<JSCExecutorFactory>(std::move(jscConfig));

Why are we keeping this file?
JSCExecutorFactory is defined in JSCExecutor.h which we are deleting with this change.

Build will pass as the compilation of this file is behind a switch but things are not cleaned-up properly. 

---
Refers to: ReactAndroid/src/main/jni/react/jni/AndroidJSCFactory.cpp:121 in d1c6edf. [](commit_id = d1c6edfe879e53a81528c2cb4258446f11c31841, deletion_comment = False)

File: ReactCommon/jsi/Android.mk [hunk 2]
```diff
@@ -41,4 +41,7 @@ else
   LOCAL_SHARED_LIBRARIES += libjsc
 endif
 
+LOCAL_SRC_FILES += $(LOCAL_V8_FILES)
+LOCAL_SHARED_LIBRARIES += libv8 libv8platform libv8base
```
Comment (rohitjain85) [inline ReactCommon/jsi/Android.mk]: This should be removed now.

**Comment by acoates-ms** [comment]
I'm going to leave the AndroidJSCFactory and full clean up to a seperate checkin -- since it probably should be done as part of something like the full jsi switch you started in https://office.visualstudio.com/ISS/_git/sdx-platform/pullrequest/230038

This change will at least remove most of the files that should have been removed from 58, and should get the build for devmain working.

**Pull request merged**